src/sys/socket/sockopt.rs

```diff
@@ -368,7 +368,7 @@ impl<'a> Set<'a, usize> for SetUsize {
     }
 
     unsafe fn ffi_ptr(&self) -> *const c_void {
```

Comment:
This shouldn't actually be `unsafe`, since they are creating raw pointers from safe references. Creating raw pointers is safe, dereferencing raw pointers is unsafe. I suggest we also change all of these functions that do that to not be unsafe.